tests/test_to_container.py

```diff
@@ -474,6 +455,23 @@ def test_enum_to_str_for_list(
         assert container == [expected]
 
 
+@mark.parametrize(
+    "cfg",
+    [
+        # to_container: throw_on_missing
+        # param(DictConfig("???"), id="dict:missing"),
+        param(DictConfig({"a": "???"}), id="dict:missing_value"),
+        param(DictConfig({"a": {"b": "???"}}), id="dict:nested"),
+        # param(ListConfig("???"), id="list:missing"),
```

Comment:
Are you testing the two commented test cases elsewhere?

Reply:
In general, do not commit commented out code. This looks like an oversight. Either remove the comments and explain why they are removed or enable the tests.

Reply:
Sorry about that. Those tests are failing because `"???" == DictConfig("???")` evaluates to `False`.
Same deal for `ListConfig`.

Here are the options I see for this:
1. submit a PR to make sure `"???" == DictConfig("???")` is `True`, and mark these cases as `xfail` for now,
2. delete these test cases, or
3. change the testing strategy to work around the problem with `DictConfig.__eq__`.

Reply:
Option 1 seems best to me because currently this equality check works with `ValueNode`s, so it seems to me that it should also be valid for containers.

Reply:
Ok, sounds good to me :)

Reply:
Ok the tests are uncommented (in https://github.com/omry/omegaconf/pull/730/commits/0df8e1a97ae7033b3679434f3544d7f4e1eeca09).